### DISCUSSION TITLE
Error using DSM Generator

### DISCUSSION THREAD

**Discussion by FPYS** [UMEP-dev/UMEP#295] [Q&A] [answered]
Hi,

I have the following error using DSM Generator : 

2021-06-23T17:21:58     WARNING    Traceback (most recent call last):
              File "C:/Users/Florian/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\UMEP\DSMGenerator\dsm_generator.py", line 643, in start_progress
              dem_array = np.array(dem_raster.ReadAsArray().astype(np.float))
             AttributeError: 'NoneType' object has no attribute 'ReadAsArray'

I've attached my files here :
[Carto.zip](https://github.com/UMEP-dev/UMEP/files/6703279/Carto.zip)

**Comment by nilswallenberg** [comment, marked as answer]
Hi,

The coordinate system for your raster file is currently in IGNF (Institut Geographique National de France) and not EPSG (European Petroleum Search Group). As of now, DSM Generator does not support IGNF, but we will update the code/plugin soon. If you are in a hurry I suggest you reproject to something in EPSG. :)

Also, your shapefile is in EPSG. Best is to have the same coordinate system in both files.

Cheers,
Nils

**Reply by FPYS** [reply]
Hi,

Thanks, I'm now able to generate a DSM but the file I get is not displayed in QGIS and I've the following error (mine is in French so it's the translation) : 

"Impossible to reproject the output extent : Impossible to create a transformation to calculate geographic north"

I guessed it has to do with reprojection on the fly but if I put my projet in the same SCR as the DSM file it doesn't show up (without the error message thought). 

I've attached the file I get : 

[DSM.zip](https://github.com/UMEP-dev/UMEP/files/6709156/DSM.zip)

**Reply by nilswallenberg** [reply]
So you need to have both your DEM (input DEM) and your shapefile (input polygon layer) in the same coordinate system. An easy way to change is to save it as a new file:

Right click the layer in the layers panel --> Export --> Save as..

Hopefully it works.
